packages/react-native/ReactCommon/react/renderer/components/image/ImageEventEmitter.cpp

```diff
@@ -13,24 +13,49 @@ void ImageEventEmitter::onLoadStart() const {
   dispatchEvent("loadStart");
 }
 
-void ImageEventEmitter::onLoad() const {
-  dispatchEvent("load");
+void ImageEventEmitter::onLoad(const ImageSource& source) const {
+  dispatchEvent("load", [source](jsi::Runtime& runtime) {
+    auto src = jsi::Object(runtime);
+    src.setProperty(runtime, "uri", source.uri);
+    src.setProperty(runtime, "width", source.size.width);
+    src.setProperty(runtime, "height", source.size.height);
+    auto payload = jsi::Object(runtime);
+    payload.setProperty(runtime, "source", src);
+    return payload;
+  });
 }
 
 void ImageEventEmitter::onLoadEnd() const {
   dispatchEvent("loadEnd");
 }
 
-void ImageEventEmitter::onProgress(double progress) const {
-  dispatchEvent("progress", [=](jsi::Runtime& runtime) {
+void ImageEventEmitter::onProgress(double progress, int64_t loaded, int64_t total) const {
+  dispatchEvent("progress", [progress, loaded, total](jsi::Runtime& runtime) {
     auto payload = jsi::Object(runtime);
     payload.setProperty(runtime, "progress", progress);
+    payload.setProperty(runtime, "loaded", (double)loaded);
+    payload.setProperty(runtime, "total", (double)total);
     return payload;
   });
 }
 
-void ImageEventEmitter::onError() const {
-  dispatchEvent("error");
+void ImageEventEmitter::onError(const ImageErrorInfo& error) const {
+  dispatchEvent("error", [error](jsi::Runtime& runtime) {
+    auto payload = jsi::Object(runtime);
+    if (error.error)
+      payload.setProperty(runtime, "error", error.error);
+    if (error.responseCode)
+      payload.setProperty(runtime, "responseCode", error.responseCode);
```

Comment:
Format nit: always wrap if in braces (not sure why clang-format isn't running in this env - it runs internally...)